### PR TITLE
add `ignore_index` option to `exact_match` metric

### PR DESCRIPTION
**Pull request by ItamarChinn**
## Fixes [1508](https://github.com/Lightning-AI/metrics/issues/1508#issue-1584796977)

This fix sets `preds` to equal the `ignore_index` in the same indices that `target == ignore_index` such that if all other elements in `preds` except the ones at indices where `target == ignore_index` are identical this constitutes an exact match.

<details>
  <summary>Before submitting</summary>

- [X] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure to **update the docs**? -> unnecessary, the issue is that the docs claim a functionality that doesn't exist
- [ ] Did you write any new **necessary tests**? -> added minimal test in the comments

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

The following test checks that it functions correctly:

```
import torch
import torchmetrics

preds = torch.tensor([[0,0,0,1,1,1,1], [1,1,1,1,1,1,1]])
target = torch.tensor([[0,0,0,1,-100,-100,-100], [0,0,1,1,-100,-100,-100]])
exact_match = torchmetrics.classification.MulticlassExactMatch(task='multiclass', num_classes=2, ignore_index=-100)
exact_match.update(preds, target)
assert exact_match.compute() == torch.tensor(0.5), "MulticlassExactMatch behaving unexpectedly" 
```

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/classification/exact_match.py [hunk 1]
```diff
@@ -124,6 +124,10 @@ def update(self, preds, target) -> None:
                 preds, target, self.num_classes, self.multidim_average, self.ignore_index
             )
         preds, target = _multiclass_stat_scores_format(preds, target, 1)
+
+        if self.ignore_index is not None:
+            preds[target == self.ignore_index] = self.ignore_index
```
Comment (SkafteNicki) [inline src/torchmetrics/classification/exact_match.py:129]: this logic should be moved to this function:
[_multiclass_stat_scores_format](https://github.com/Lightning-AI/metrics/blob/f075e6a28e6f1a7e095968ecf0a1da28a5c8aa3c/src/torchmetrics/functional/classification/exact_match.py#L40-L49)

**Review by SkafteNicki — CHANGES_REQUESTED** [review]

**Review by SkafteNicki — APPROVED** [review]

**Comment by championsnet** [comment]
This needs fixing for multilabel exact match as well
